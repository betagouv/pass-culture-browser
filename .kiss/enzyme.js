import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
