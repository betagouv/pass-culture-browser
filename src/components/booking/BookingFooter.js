/* eslint
  react/jsx-one-expression-per-line: 0 */
import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

const BookingFooter = ({
  onCancel,
  onSubmit,
  isBooked,
  isSubmitting,
  canSubmitForm,
}) => {
  const showOkButton = isBooked
  const showCancelButton = !isSubmitting && !isBooked
  const showSubmitButton = showCancelButton && canSubmitForm
  return (
    <Fragment>
      {showCancelButton && (
        <button type="reset" onClick={onCancel} className="text-center my5">
          <span>Annuler</span>
        </button>
      )}
      {showSubmitButton && (
        <button type="submit" onClick={onSubmit} className="text-center my5">
          <b>Valider</b>
        </button>
      )}
      {showOkButton && (
        <button type="button" onClick={onCancel} className="text-center my5">
          <b>OK</b>
        </button>
      )}
    </Fragment>
  )
}

BookingFooter.defaultProps = {}

BookingFooter.propTypes = {
  canSubmitForm: PropTypes.bool.isRequired,
  isBooked: PropTypes.bool.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
}

export default BookingFooter
