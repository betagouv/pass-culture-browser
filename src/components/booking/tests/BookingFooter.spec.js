// jest --env=jsdom ./src/components/booking/tests/BookingFooter --watch
import React from 'react'
import { shallow } from 'enzyme'

import BookingFooter from '../BookingFooter'

const onCancelMock = jest.fn()
const onSubmitMock = jest.fn()

describe('src | components | pages | search | BookingFooter', () => {
  describe('snapshot', () => {
    it('can not submit, is not booked, is not submitting', () => {
      // given
      const props = {
        canSubmitForm: false,
        isBooked: false,
        isSubmitting: false,
        onCancel: onCancelMock,
        onSubmit: onSubmitMock,
      }

      // when
      const wrapper = shallow(<BookingFooter {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
