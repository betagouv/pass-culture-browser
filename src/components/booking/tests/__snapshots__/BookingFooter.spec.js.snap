// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | search | BookingFooter snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookingFooter
    canSubmitForm={true}
    isBooked={true}
    isSubmitting={true}
    onCancel={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        false,
        <button
          className="text-center my5"
          onClick={[MockFunction]}
          type="button"
        >
          <b>
            OK
          </b>
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <b>
            OK
          </b>,
          "className": "text-center my5",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OK",
          },
          "ref": null,
          "rendered": "OK",
          "type": "b",
        },
        "type": "button",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          false,
          <button
            className="text-center my5"
            onClick={[MockFunction]}
            type="button"
          >
            <b>
              OK
            </b>
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              OK
            </b>,
            "className": "text-center my5",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OK",
            },
            "ref": null,
            "rendered": "OK",
            "type": "b",
          },
          "type": "button",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
